From 3e2a03b4c6ad39149986f8fddd99dc759a734a3e Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:40:02 -0400
Subject: [net] bridge: fdb dumping takes a filter device

Message-id: <1409658042-22689-242-git-send-email-sassmann@redhat.com>
Patchwork-id: 90421
O-Subject: [RHEL7.1 PATCH 241/281] bridge: fdb dumping takes a filter device
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jamal Hadi Salim <jhs@mojatatu.com>

Dumping a bridge fdb dumps every fdb entry
held. With this change we are going to filter
on selected bridge port.

Signed-off-by: Jamal Hadi Salim <jhs@mojatatu.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 5d5eacb34c9e1fdc0a47b885d832eaa4de860dc7)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index f06097c..36c8eaf 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -7095,13 +7095,14 @@ static int i40e_ndo_fdb_del(struct ndmsg *ndm,
 static int i40e_ndo_fdb_dump(struct sk_buff *skb,
         struct netlink_callback *cb,
         struct net_device *dev,
+        struct net_device *filter_dev,
         int idx)
 {
  struct i40e_netdev_priv *np = netdev_priv(dev);
  struct i40e_pf *pf = np->vsi->back;
 
  if (pf->flags & I40E_FLAG_SRIOV_ENABLED)
-  idx = ndo_dflt_fdb_dump(skb, cb, dev, idx);
+  idx = ndo_dflt_fdb_dump(skb, cb, dev, filter_dev, idx);
 
  return idx;
 }
-- 
1.7.1